### PR TITLE
feat(usefullscreen): pass requestFullscreenOptions

### LINKED ISSUE #505: Missing fullscreen options
**Describe the bug**
The fullscreen hook options are missing [navigationUI](https://developer.mozilla.org/en-US/docs/Web/API/FullscreenOptions/navigationUI)
```ts
declare type FullScreenOptions = {
    onChange?: OnChangeEventCallback;
    onError?: EventCallback;
};
```

**Expected behavior**
[FullscreenOptions](https://developer.mozilla.org/en-US/docs/Web/API/FullscreenOptions) should be supported

**Additional context**
<details><summary>not working</summary>

As a temporary workaround, we can cast an overridded type to useFullscreen.

```ts
type ExtendedFullscreenOptions = Parameters<typeof useFullscreen>[0] & {
  // https://developer.mozilla.org/en-US/docs/Web/API/FullscreenOptions/navigationUI
  navigationUI?: "hide" | "show" | "auto";
};
interface IuseFullscreen {
  (options?: ExtendedFullscreenOptions): ReturnType<typeof useFullscreen>;
}

const MyComponent: React.FC = () => {
  const fullscreenApi = (useFullscreen as IuseFullscreen)({
    navigationUI: "hide", // Hide the Android bottom menu
  });

  return;
}
```

</details>

Options should be passed to requestFullscreen, but rooks isn't passing anything
https://github.com/imbhargav5/rooks/blob/main/src/hooks/useFullscreen.ts#L162

### PR DESCRIPTION
**Pull request by phmngocnghia**
fix #505

### CODE REVIEW

**Comment by imbhargav5** [comment]
@AdrienLemaire would you like to review this one?

File: src/__tests__/useFullscreen.spec.tsx [hunk 1]
```diff
@@ -1,12 +1,26 @@
...
+    (document as any).exitFullscreen = jest.fn();
+    const element = { requestFullscreen: jest.fn() };
+    const mockRequestFullscreenOptions = { navigationUI: "show" };
+    const { result } = renderHook(() =>
+      useFullscreen({ requestFullscreenOptions: { navigationUI: "show" } })
+    );
+    act(() => {
+      result.current?.request(element as any);
+    });
+    expect(element.requestFullscreen).toHaveBeenCalledWith(
```
Comment (imbhargav5) [inline src/__tests__/useFullscreen.spec.tsx:22]: Excellent job @phmngocnghia !

File: src/__tests__/useFullscreen.spec.tsx [hunk 2]
```diff
@@ -1,12 +1,26 @@
...
     expect(useFullscreen).toBeDefined();
   });
-  it('should return an object', () => {
+  it("should return an object", () => {
     const { result } = renderHook(() => useFullscreen());
-    expect(typeof result.current).toBe('object');
+    expect(typeof result.current).toBe("object");
+  });
+  it.only("should forward requestFullscreenOptions to requestFullscreen", () => {
+    (document as any).exitFullscreen = jest.fn();
```
Comment (imbhargav5) [inline src/__tests__/useFullscreen.spec.tsx:13]: Do we need to assert `any` here? 🤔 Curious as to why this is needed.

**Review by imbhargav5 — APPROVED** [review]

File: src/__tests__/useFullscreen.spec.tsx (hunk 2 above)
Comment (phmngocnghia) [reply src/__tests__/useFullscreen.spec.tsx]: Previous, I put `globals as any`, but forgot to remove any when I changed to `document`

Comment (phmngocnghia) [reply src/__tests__/useFullscreen.spec.tsx]: I removed `any`

**Pull request merged**